docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,320 @@
+                    INFO     Subtask a88c0feeaef6493796a9148ed68c9caf
+                             Response: 111.1080555135405
+[04/26/24 13:06:49] INFO     ToolkitTask 36b9dea13b9d479fb752014f41dca54c
+                             Output: The square root of 12345 is approximately 111.108.
+```
+
+Since the result of the Calculator Tool is neither sensitive nor too large, we can set `off_prompt` to `False` and not use Task Memory.
+
+Let's explore what happens when `off_prompt` is set to `True`:
+
+```python
+from griptape.structures import Agent
+from griptape.tools import Calculator
+
+# Create an agent with the Calculator tool
+agent = Agent(
+    tools=[Calculator(off_prompt=True)]
+)
+
+agent.run("What is 10 raised to the power of 5?")
+```
+
+
+```
+[04/26/24 13:07:02] INFO     ToolkitTask ecbb788d9830491ab72a8a2bbef5fb0a
+                             Input: What is the square root of 12345?
+[04/26/24 13:07:10] INFO     Subtask 4700dc0c2e934d1a9af60a28bd770bc6
+                             Thought: To find the square root of a number, we can use the Calculator action with the expression "sqrt(12345)". However, the Calculator
+                             action only supports basic arithmetic operations and does not support the sqrt function. Therefore, we need to use the equivalent expression
+                             for square root which is raising the number to the power of 0.5.
+                             Actions: [{"name": "Calculator", "path": "calculate", "input": {"values": {"expression": "12345**0.5"}}, "tag": "sqrt_calculation"}]
+                    INFO     Subtask 4700dc0c2e934d1a9af60a28bd770bc6
+                             Response: Output of "Calculator.calculate" was stored in memory with memory_name "TaskMemory" and artifact_namespace
+                             "6be74c5128024c0588eb9bee1fdb9aa5"
+[04/26/24 13:07:16] ERROR    Subtask ecbb788d9830491ab72a8a2bbef5fb0a
+                             Invalid action JSON: Or({Literal("name", description=""): 'Calculator', Literal("path", description="Can be used for computing simple
+                             numerical or algebraic calculations in Python"): 'calculate', Literal("input", description=""): {'values': Schema({Literal("expression",
+                             description="Arithmetic expression parsable in pure Python. Single line only. Don't use variables. Don't use any imports or external
+                             libraries"): <class 'str'>})}, Literal("tag", description="Unique tag name for action execution."): <class 'str'>}) did not validate {'name':
+                             'Memory', 'path': 'get', 'input': {'memory_name': 'TaskMemory', 'artifact_namespace': '6be74c5128024c0588eb9bee1fdb9aa5'}, 'tag':
+                             'get_sqrt_result'}
+                             Key 'name' error:
+                             'Calculator' does not match 'Memory'
```

Comment:
Should we truncate this as well and just say "errors are generated" or something?

Reply:
We should probably leave the actual error message the customer would see in the example, but fine with truncating any of the middle.

Reply:
I think it's important to show a full error message, I already truncated quite a bit of other output.